### ISSUE TITLE
Why there needs a PAD?

### ISSUE THREAD

**Issue by standbyme** [gaussic/text-classification-cnn-rnn#77]
https://github.com/gaussic/text-classification-cnn-rnn/blob/155e27b4dac6fedca9e5ed909fd2465a2d54ea74/data/cnews_loader.py#L70

I have refered to 3 websites which you have mentioned but fail to find the reason.
Look forward to your reply😆
Thanks

**Comment by gaussic** [comment]
To answer your two questions in this issue:

As we can see, we use keras's `pad_sequences` to pad the sequences into fixed length:

https://github.com/gaussic/text-classification-cnn-rnn/blob/155e27b4dac6fedca9e5ed909fd2465a2d54ea74/data/cnews_loader.py#L110

And in [keras pad_sequences](https://keras.io/preprocessing/sequence/#pad_sequences), the padding value is expected to be 0.0, which means 0 is reserved as a padding index. Therefore, we use `<PAD>` to reserve the 0 index.

And to make the `vocab_size` equal to 5000, we use most `vocab_size - 1` words and the `<PAD>` sign as the final vocabulary.

**Comment by standbyme** [comment]
Thanks a lot😆
